src/embedding/ad.ts

```diff
@@ -0,0 +1,584 @@
+// (C) 2020 GoodData Corporation
+
+import isEmpty = require('lodash/isEmpty');
+import {
+    CommandFailed,
+    IObjectMeta,
+    IGdcMessageEvent,
+    IGdcMessageEventDataWrapper,
+    ICommandFailedBody
+} from './common';
+
+type ANALYTICAL_DESIGNER = 'analyticalDesigner';
+const AD_PRODUCT_NAME = 'analyticalDesigner';
+
+// TODO: these are part of client (gooddata-js), move to model (typings)
+export interface IBaseExportConfig {
+    title?: string;
+    format?: 'xlsx' | 'csv' | 'raw';
+    mergeHeaders?: boolean;
+}
+
+// TODO: these are part of client (gooddata-js), move to model (typings)
+// NOTE: AFM is ommitted on purpose; it should be added by AD itself; create new type using Omit<>
+export interface IExportConfig extends IBaseExportConfig {
+    showFilters?: boolean;
+}
+
+export namespace EmbeddedAnalyticalDesigner {
+    export type IADMessageEvent<T, TBody> = IGdcMessageEvent<ANALYTICAL_DESIGNER, T, TBody>;
+
+    export enum GdcAdCommandType {
+        Save = 'adSave',
+        SaveAs = 'adSaveAs',
+        Export = 'adExport',
+        Undo = 'adUndo',
+        Redo = 'adRedo'
+    }
```

Comment:
I think it should be
```
export enum GdcAdCommandType {
        Save = 'saveInsight',
        SaveAs = 'saveAsInsight',
        Export = 'exportInsight',
        Undo = 'undo',
        Redo = 'redo'
    }
```
because we already have `openInsight`, so we should follow this style

Reply:
WE should follow camel case as `openInsight`

Reply:
updated